### PR TITLE
[WIP] fix TPR single group, multiple versions issue

### PR DESCRIPTION
**Pull request by adohe-zz**
Ref [here](https://github.com/kubernetes/features/issues/95) for more detail. @deads2k @brendandburns ptal.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36157)

### CODE REVIEW

File: pkg/master/thirdparty/thirdparty.go [hunk 1]
```diff
@@ -142,11 +144,38 @@ func (m *ThirdPartyResourceServer) RemoveThirdPartyResource(path string) error {
...
+		}{}
+		routes := ws.Routes()
+		for _, route := range routes {
+			match, _ := regexp.MatchString(pattern, route.Path)
+			if match {
+				routesToRemove = append(routesToRemove, struct{ method, path string }{route.Method, route.Path})
+			}
+		}
+
+		for _, routeToeRemove := range routesToRemove {
```
Comment (deads2k) [inline pkg/master/thirdparty/thirdparty.go]: typo.  Though ToeRemoval sounds interesting..... :)

File: pkg/registry/extensions/rest/thirdparty_controller.go [hunk 2]
```diff
@@ -126,6 +133,13 @@ func MakeThirdPartyPath(group string) string {
 	return thirdpartyprefix + "/" + group
 }
 
+func getFullThirdPartyPath(group, resource string) string {
+	if len(group) == 0 {
+		return thirdpartyprefix
+	}
+	return thirdpartyprefix + "/" + group + "/" + resource
```
Comment (deads2k) [inline pkg/registry/extensions/rest/thirdparty_controller.go]: This path doesn't look right for a versioned API.

Comment (AdoHe) [inline pkg/registry/extensions/rest/thirdparty_controller.go]: @deads2k yes, I am thinking about this PR together with the second issue: `Multiple Versions - it's currently impossible to add more than one version for a group`

**Comment by AdoHe** [comment]
Close this in favor of #36977

**Pull request closed without merging**